### PR TITLE
Improve ruleset templates

### PR DESCRIPTION
**Pull request by hlysine**
Several changes to the ruleset templates:

1. Add `<NoDefaultExcludes>true</NoDefaultExcludes>` in the ruleset project so that files starting with a dot (`.editorconfig`, `.gitignore`) can be included in the templates.

2. Update `.editorconfig`, `.gitignore` and DotSettings in templates to match those in root directory.

3. Changes assembly titles from `Sample` to the template names so that they can be replaced by `dotnet new -n <name>`.

### CODE REVIEW

File: Templates/Rulesets/ruleset-empty/.editorconfig [hunk 1]
```diff
@@ -197,4 +189,6 @@ dotnet_diagnostic.IDE0069.severity = none
 dotnet_diagnostic.CA2225.severity = none
 
 # Banned APIs
-dotnet_diagnostic.RS0030.severity = error
\ No newline at end of file
+dotnet_diagnostic.RS0030.severity = error
+
+dotnet_diagnostic.OLOC001.license_header = // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.\n// See the LICENCE file in the repository root for full licence text.
```
Comment (peppy) [inline Templates/Rulesets/ruleset-empty/.editorconfig:194]: We don't want this in the templates.

File: Templates/Rulesets/ruleset-example/.editorconfig [hunk 2]
```diff
@@ -197,4 +189,6 @@ dotnet_diagnostic.IDE0069.severity = none
 dotnet_diagnostic.CA2225.severity = none
 
 # Banned APIs
-dotnet_diagnostic.RS0030.severity = error
\ No newline at end of file
+dotnet_diagnostic.RS0030.severity = error
+
+dotnet_diagnostic.OLOC001.license_header = // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.\n// See the LICENCE file in the repository root for full licence text.
```
Comment (peppy) [inline Templates/Rulesets/ruleset-example/.editorconfig:194]: As above.

File: Templates/Rulesets/ruleset-scrolling-empty/.editorconfig [hunk 3]
```diff
@@ -197,4 +189,6 @@ dotnet_diagnostic.IDE0069.severity = none
 dotnet_diagnostic.CA2225.severity = none
 
 # Banned APIs
-dotnet_diagnostic.RS0030.severity = error
\ No newline at end of file
+dotnet_diagnostic.RS0030.severity = error
+
+dotnet_diagnostic.OLOC001.license_header = // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.\n// See the LICENCE file in the repository root for full licence text.
```
Comment (peppy) [inline Templates/Rulesets/ruleset-scrolling-empty/.editorconfig:194]: Also this.

File: Templates/Rulesets/ruleset-scrolling-example/.editorconfig [hunk 4]
```diff
@@ -197,4 +189,6 @@ dotnet_diagnostic.IDE0069.severity = none
 dotnet_diagnostic.CA2225.severity = none
 
 # Banned APIs
-dotnet_diagnostic.RS0030.severity = error
\ No newline at end of file
+dotnet_diagnostic.RS0030.severity = error
+
+dotnet_diagnostic.OLOC001.license_header = // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.\n// See the LICENCE file in the repository root for full licence text.
```
Comment (peppy) [inline Templates/Rulesets/ruleset-scrolling-example/.editorconfig:194]: You get the idea.

**Review by peppy — CHANGES_REQUESTED** [review]

File: Templates/Rulesets/ruleset-empty/.editorconfig (hunk 1 above)
Comment (hlysine) [reply Templates/Rulesets/ruleset-empty/.editorconfig]: Oops sorry. Should've checked after copy-pasting.

**Review by frenzibyte — APPROVED** [review]
On diffing and testing, everything seems to check out. One thing I'm not sure about is the presence of the copyright license header across the ruleset templates projects, but I suppose that's for another PR.

**Review by peppy — APPROVED** [review]

**Pull request merged**